src/bin/set-version/main.rs

```diff
@@ -0,0 +1,271 @@
+            cmd.manifest_path(path);
+        }
+        let result = cmd.exec().map_err(|e| {
+            Error::from(e.compat()).chain_err(|| "Failed to get workspace metadata")
+        })?;
+        result
+            .packages
+            .into_iter()
+            .map(|package| {
+                Ok((
+                    LocalManifest::try_new(Path::new(&package.manifest_path))?,
+                    package,
+                ))
+            })
+            .collect::<Result<Vec<_>>>()
+            .map(Manifests)
+    }
+
+    fn get_pkgid(pkgid: &str) -> Result<Self> {
+        let package = manifest_from_pkgid(pkgid)?;
+        let manifest = LocalManifest::try_new(Path::new(&package.manifest_path))?;
+        Ok(Manifests(vec![(manifest, package)]))
+    }
+
+    /// Get the manifest specified by the manifest path. Try to make an educated guess if no path is
+    /// provided.
+    fn get_local_one(manifest_path: &Option<PathBuf>) -> Result<Self> {
+        let resolved_manifest_path: String = find(&manifest_path)?.to_string_lossy().into();
+
+        let manifest = LocalManifest::find(&manifest_path)?;
+
+        let mut cmd = cargo_metadata::MetadataCommand::new();
+        cmd.no_deps();
+        if let Some(path) = manifest_path {
+            cmd.manifest_path(path);
+        }
+        let result = cmd
+            .exec()
+            .map_err(|e| Error::from(e.compat()).chain_err(|| "Invalid manifest"))?;
+        let packages = result.packages;
+        let package = packages
+            .iter()
+            .find(|p| p.manifest_path.to_string_lossy() == resolved_manifest_path)
+            // If we have successfully got metadata, but our manifest path does not correspond to a
+            // package, we must have been called against a virtual manifest.
+            .chain_err(|| {
+                "Found virtual manifest, but this command requires running against an \
+                 actual package in this workspace. Try adding `--workspace`."
+            })?;
+
+        Ok(Manifests(vec![(manifest, package.to_owned())]))
+    }
+
+    /// Update the manifests on disk with the given version.
+    fn set_version(self, dry_run: bool, version_or_bump: VersionOrBump) -> Result<()> {
+        if dry_run {
+            dry_run_message()?;
+        }
+        for (mut manifest, package) in self.0 {
+            let version = match version_or_bump {
+                VersionOrBump::Version(ref version) => version.clone(),
+                VersionOrBump::Bump(bump) => {
+                    let mut version = package.version.clone();
+                    match bump {
+                        Bump::Major => version.increment_major(),
+                        Bump::Minor => version.increment_minor(),
+                        Bump::Patch => version.increment_patch(),
+                    }
+                    version
+                }
+            };
+            println!("{}: {} -> {}", package.name, package.version, version);
```

Comment:
it'd be good to print with colors like we do with cargo upgrade to be more consistent

Reply:
@ordian Yeah, makes sense, thanks for reviewing! :)
@hdevalke, do you still have interest in merging this PR? I see you use `convco` together with `cargo-release`, which seems to implement similar functionality, so it might be reasonable to keep them separate. What do you guys think?

Reply:
For me it is fine to use cargo release. I did not had the need to change the version without it.